### ISSUE TITLE
MNT: python3.11 breaks Iris doc build

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#5232]
## 📰 Custom Issue

We have a bespoke Sphinx extension that [uses `inspect.getargspec` and `inspect.formatargspec`](https://github.com/rcomer/iris/blob/09914f76df3e1d1fb8dfdb6b64ca23730a1c335a/docs/src/sphinxext/custom_class_autodoc.py#L72-L82).  Neither of these function exist any more at python 3.11.  The obvious replacement for `getargspec` is `getfullargspec` but I'm not clear what to do about `formatargspec`.

I also note that this extension has been here since the initial commit, so seems reasonable to wonder if there are now off-the-shelf solutions for whatever it's for.

**Comment by tkknight** [comment]
I played with removing all of the custom sphinx extensions in https://github.com/SciTools/iris/tree/main/docs/src/sphinxext in favor of sphinx-apidoc.

I believe it worked but then I shelved it due to other priorities.  One likely impact thought is the API docs naming or directory structure may change meaning switching between versions of the docs when viewing the API docs may break.

I will take a look at this sometime soon.
